ember-cli-flash/src/flash/object.js

```diff
@@ -1,22 +1,22 @@
-import Evented from '@ember/object/evented';
-import EmberObject, { set } from '@ember/object';
 import { cancel, later } from '@ember/runloop';
-import { guidFor } from '../utils/computed';
+import { tracked } from '@glimmer/tracking';
+import { guidFor } from '@ember/object/internals';
 
-// Note:
-// To avoid https://github.com/adopted-ember-addons/ember-cli-flash/issues/341 from happening, this class can't simply be called Object
-export default class FlashObject extends EmberObject.extend(Evented) {
+export default class FlashObject {
   exitTimer = null;
-  exiting = false;
+  @tracked exiting = false;
   isExitable = true;
   initializedTime = null;
+  isDestroyed = false;
 
-  @(guidFor('message').readOnly())
-  _guid;
+  get _guid() {
+    return guidFor(this.message?.toString());
+  }
 
-  // eslint-disable-next-line ember/classic-decorator-hooks
-  init() {
-    super.init(...arguments);
+  constructor(messageOptions) {
+    for (const [key, value] of Object.entries(messageOptions)) {
+      this[key] = value;
+    }
```

Comment:
Should possibly only assign allow-listed keys 🤷🏻 or even enumerate the specific keys to assign 🤷🏻

Reply:
we'll probably want to disallow this at some point
because we can pass components as values (and should) and migrate away from string-based component references.

Reply:
Ah – we should update the example, but that's just demonstrating that _any_ arbitrary keys can be set to any value.